Source/Dafny/Compiler-Csharp.cs

```diff
@@ -1709,11 +1721,41 @@ protected class ClassWriter : IClassWriter {
     }
 
     protected override void EmitSeqConstructionExpr(SeqConstructionExpr expr, bool inLetExprBody, TargetWriter wr) {
-      wr.Write("{0}<{1}>.Create(", DafnySeqClass, TypeName(expr.Type.AsSeqType.Arg, wr, expr.tok));
-      TrExpr(expr.N, wr, inLetExprBody);
-      wr.Write(", ");
-      TrExpr(expr.Initializer, wr, inLetExprBody);
-      wr.Write(")");
+      var elmtType = expr.Type.AsSeqType.Arg;
```

Comment:
Could we write a private method with the `LambdaExpr` code with a comment saying that this is equivalent but faster than the corresponding generic code, and then call that here?

Reply:
Sure, done.